hazelcast/src/main/java/com/hazelcast/client/impl/clientside/ClientLoggingService.java

```diff
@@ -86,6 +86,21 @@ public ILogger getLogger(@Nonnull Class clazz) {
         return getOrPutIfAbsent(mapLoggers, clazz.getName(), loggerConstructor);
     }
 
+    @Override
+    public void removeLogger(@Nonnull String name) {
+        checkNotNull(name, "name must not be null");
+        loggerFactory.removeLogger(name);
```

Comment:
Does this need to be synchronized with concurrent calls to `getLogger`? I can see all calls to `getLogger` are thread-safe as they use striped locks but `removeLogger` here does not coordinate with methods adding loggers.

Reply:
Good catch.

I think, the simplest way to fix this problem is to use the fact that `loggerFactory` accesses are guarded with CHM (`mapLoggers`) operations and we could use CHM's atomicity guarantees for the deletion:
```java
    @Override
    public void removeLogger(@Nonnull String name) {
        checkNotNull(name, "name must not be null");
        mapLoggers.computeIfPresent(name, (k, v) -> {
            loggerFactory.removeLogger(name);
            // delete the entry
            return null;
        });        
    }
```